Prox/Prox/Models/AppState.swift

```diff
@@ -6,19 +6,37 @@ import Foundation
 
 class AppState {
     enum State: String {
-        case loading, details, carousel, background, exiting, unknown
+        case initial, loading, permissions, details, carousel, background, unknown
     }
 
     private static var cardVisits = Set<Int>()
 
     // Initial app state
-    private static var state = State.loading
+    private static var state = State.initial
     private static var preBackgroundState: State?
 
     static func getState() -> State {
         return state
     }
 
+    static func enterLoading() {
+        updateSessionState(newState: State.loading)
+    }
+
+    static func requestPermissions() {
+        if (state == State.loading) {
```

Comment:
Why do we sometimes consider permissions a session and sometimes consider permissions an event? I'm concerned it'd be hard to accurately track permissions requests this way.

Also, if we're displaying a dialog, I think we'd still be loading under the hood so I'd think we should continue the loading session (fwiw, I don't think we do any cleanup in `ResignActive` or `EnterBackground` so I think we'd continue loading during our ~5s background execution time too) – why do we cancel the loading session here?